src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/SPDXExtensions.cs

```diff
@@ -19,158 +19,130 @@ namespace Microsoft.Sbom.Parsers.Spdx30SbomParser.Utils;
+        if (fileInfo.Checksum is null || !fileInfo.Checksum.Any(c => c.Algorithm == AlgorithmName.SHA1))
         {
-            throw new MissingHashValueException($"The file {fileName} is missing the {HashAlgorithmName.SHA1} hash value.");
+            throw new MissingHashValueException($"The file {fileInfo.Path} is missing the {HashAlgorithmName.SHA1} hash value.");
         }
 
         // Get the SHA1 for this file.
-        var sha1Value = checksums.Where(c => c.Algorithm == AlgorithmName.SHA1)
+        var sha1Value = fileInfo.Checksum.Where(c => c.Algorithm == AlgorithmName.SHA1)
             .Select(s => s.ChecksumValue)
             .FirstOrDefault();
 
-        return sha1Value;
+        element.SpdxId = CommonSPDXUtils.GenerateSpdxFileId(fileInfo.Path, sha1Value);
+        return element.SpdxId;
     }
 
     /// <summary>
-    /// Adds ExternalSpdxId property to the SPDXExternalDocumentReference based on name and checksum information.
+    /// Adds SPDX ID that corresponds to the package info.
     /// </summary>
-    public static void AddExternalSpdxId(this ExternalMap reference, string name, IEnumerable<Checksum> checksums)
+    /// <param name="spdxPackage"></param>
+    /// <param name="packageInfo"></param>
+    /// <returns>Package ID that encapsulates unique info about a package.</returns>
+    public static string AddSpdxId(this Package spdxPackage, SbomPackage packageInfo)
     {
-        if (reference is null)
-        {
-            throw new ArgumentNullException(nameof(reference));
-        }
-
-        if (string.IsNullOrEmpty(name))
+        if (packageInfo is null)
         {
-            throw new ArgumentException($"'{nameof(name)}' cannot be null or empty.", nameof(name));
+            throw new ArgumentNullException(nameof(packageInfo));
         }
 
-        var sha1checksums = checksums.Where(c => c.Algorithm == AlgorithmName.SHA1);
-        if (checksums is null || !sha1checksums.Any())
+        // Get package identity as package name and package version. If version is empty, just use package name
+        var packageIdentity = $"{packageInfo.Type}-{packageInfo.PackageName}";
+        if (!string.IsNullOrWhiteSpace(packageInfo.PackageVersion))
         {
-            throw new MissingHashValueException($"The external reference {name} is missing the {HashAlgorithmName.SHA1} hash value.");
+            packageIdentity = string.Join("-", packageInfo.Type, packageInfo.PackageName, packageInfo.PackageVersion);
         }
 
-        // Get the SHA1 for this file.
-        var sha1Value = sha1checksums.FirstOrDefault().ChecksumValue;
+        spdxPackage.SpdxId = CommonSPDXUtils.GenerateSpdxPackageId(packageInfo.Id ?? packageIdentity);
+        return spdxPackage.SpdxId;
+    }
 
-        reference.ExternalSpdxId = GenerateSpdxExternalDocumentId(name, sha1Value);
+    public static void AddSpdxId(this CreationInfo creationInfo)
```

Comment:
The reordering of these methods makes it a lot harder to compare the old version to the new version. Is it possible to restore to the original order to simplify the review?

Reply:
I've reordered them, could you take a look and see if that helps?